tests/ga/test_policy_engine.py

```diff
@@ -15,20 +15,374 @@
 # Requires Python 2.4+ and Openssl 1.0+
 #
 
+import os
+import json
+
 from tests.lib.tools import AgentTestCase
-from azurelinuxagent.ga.policy.policy_engine import PolicyEngine
+from azurelinuxagent.ga.policy.policy_engine import ExtensionPolicyEngine
 from tests.lib.tools import patch
+from azurelinuxagent.common.protocol.restapi import Extension
+
+TEST_EXTENSION_NAME = "Microsoft.Azure.ActiveDirectory.AADSSHLoginForLinux"
+
+
+class TestExtensionPolicyEngine(AgentTestCase):
```

Comment:
I think we should separate these unit tests into separate classes. 
TestPolicyEngine
TestExtensionPolicyEngine

policy enabled/parsing logic should be in TestPolicyEngine